### PR TITLE
RHIROS-426 | Upgrade ROS api routes from v0 to v1

### PR DESCRIPTION
**Pull request by saltgen**
### Changes:
1. Replaced "v0" with "v1" on all routes
2. Modified usages references of all routes accordingly
3. Renamed `v0` directory in ros.api to `v1`

### Reference links:
1. [consoledot](https://consoledot.pages.redhat.com/docs/dev/developer-references/rest/versioning.html#_semantic_versioning)
2. [semver spec](https://semver.org/#spec-item-5)

### Additional:

Added tests for api endpoints

### Note:
Wrote test for the `status` endpoint, working on tests for other endpoints as well, hence MR is on WIP
Tried my best to update all references to "v1", in case I missed any please let me know

### CODE REVIEW

File: local-ros-backend.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module.exports.routes = {
-    '/api/ros/v0' : { host: 'http://localhost:8080' },
+    '/api/ros/v1' : { host: 'http://localhost:8080' },
```
Comment (apuntamb) [inline local-ros-backend.js:2]:
```suggestion
    '/api/ros/v1' : { host: 'http://localhost:8000' },
```

**Review by apuntamb — CHANGES_REQUESTED** [review]
Thanks for the PR @saltgen! Looks good. Could you check if this version might need a bump too https://github.com/RedHatInsights/ros-backend/blob/main/ros/openapi/openapi.json#L11 ?

**Review by r14chandra — CHANGES_REQUESTED** [review]
Need to change v0 on frontend repo also, here also https://github.com/RedHatInsights/ros-frontend/blob/main/src/constants.js#L2

File: local-ros-backend.js (hunk 1 above)
Comment (saltgen) [reply local-ros-backend.js:2]: Yes that line did need "v1.0", change pushed, also I believe we will need to raise and MR to update [here](https://github.com/RedHatInsights/cloud-services-config/blob/3bc11662bbc9570f19b0e6670e84b045a83ee41f/main.yml#L922)

Respectively, do we need to make changes on 3Scale?

**Comment by saltgen** [comment]
> Need to change v0 on frontend repo also, here also https://github.com/RedHatInsights/ros-frontend/blob/main/src/constants.js#L2

@r14chandra Good point!
@PreetiW Does front-end need any more changes in respect to v1?

**Comment by apuntamb** [comment]
@saltgen can you resolve the conflicts before we merge this?

**Comment by saltgen** [comment]
> @saltgen can you resolve the conflicts before we merge this?

@apuntamb Ah I think these conflicts came up, cause of the earlier merges, on it.

**Comment by saltgen** [comment]
> @saltgen can you resolve the conflicts before we merge this?

@apuntamb Conflicts resolved :+1:

File: local-ros-backend.js (hunk 1 above)
Comment (kgaikwad) [reply local-ros-backend.js:2]: @saltgen  - After these changes, you will need to fix URL paths used inside templates - https://github.com/RedHatInsights/e2e-deploy/tree/master/templates/ros

Example - https://github.com/RedHatInsights/e2e-deploy/blob/master/templates/ros/ros-backend.yml#L107

**Review by kgaikwad — APPROVED** [review]
LGTM

**Pull request merged**